frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -373,6 +423,66 @@ function filterNullDimensionValues(dataset: ChartDataset) {
   return filteredDataset;
 }
 
+const Y_AXIS_CROSSING_ERROR = Error(
+  t`Y-axis must not cross 0 when using log scale.`,
+);
+
+export function replaceZeroesForLogScale(
+  dataset: ChartDataset,
+  seriesDataKeys: DataKey[],
+) {
+  let hasZeros = false;
+  let minNonZeroValue = Infinity;
+  let sign: number | undefined = undefined;
+
+  dataset.forEach(datum => {
+    const datumNumericValues = seriesDataKeys
+      .map(key => getNumberOr(datum[key], null))
+      .filter(isNotNull);
+
+    const hasPositive = datumNumericValues.some(value => value > 0);
+    const hasNegative = datumNumericValues.some(value => value < 0);
+
+    if (hasPositive && hasNegative) {
+      throw Y_AXIS_CROSSING_ERROR;
```

Comment:
The old code threw strings, let's change it to throwing errors. Afair the code that handles it supports showing error messages from `error.message`

Reply:
Aren't we already throwing an error here? Y_AXIS_CROSSING_ERROR is an error object, not a string

Reply:
My brain glitched, never mind 👍